content/techniques/mongo.md

```diff
@@ -173,6 +171,94 @@ export class CatsService {
 }
 ```
 
+#### Hooks (middleware)
+
+Middleware (also called pre and post hooks) are functions which are passed control during execution of asynchronous functions. Middleware is specified on the schema level and is useful for writing plugins ([source](https://mongoosejs.com/docs/middleware.html)). Calling `pre()` or `post()` after compiling a model does not work in Mongoose. In order to register a hook **before** model registration, use the `forFeatureAsync()` method.
+
+```typescript
+@Module({
+  imports: [
+    MongooseModule.forFeatureAsync([
+      {
+        name: 'Cat',
+        useFactory: () => {
+          const schema = CatsSchema;
+          schema.pre('save', () => console.log('Hello from pre save'));
+          return schema;
+        },
+      },
+    ]),
+  ],
+})
+export class AppModule {}
+```
+
+Like other [factory providers](https://docs.nestjs.com/fundamentals/custom-providers#factory-providers-usefactory), our factory function can be `async` and can inject dependencies through `inject`.
+
+```typescript
+@Module({
+  imports: [
+    MongooseModule.forFeatureAsync([
+      {
+        name: 'Cat',
+        import [ConfigModule],
+        useFactory: (configService: ConfigService) => {
+          const schema = CatsSchema;
+          schema.pre('save', () =>
+            console.log(`${configService.getString('APP_NAME')}: Hello from pre save`),
+          );
+          return schema;
+        },
+        inject: [ConfigService],
+      }
+    ]),
+  ],
+})
+export class AppModule {}
+```
+
+#### Plugins
+
+To register a [plugin](https://mongoosejs.com/docs/plugins.html) for a given schema, use the `forFeatureAsync()` method.
```

Comment:
Suggestion: add the following:

As with `pre()` and `post()` hooks, use the `useFactory` technique to access a schema in order to invoke the `plugin()` method.